core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -31,32 +31,28 @@ public HashPartitionStrategy(final HashPartitionStrategyCrutch hashPartitioningC
     }
 
     @Override
-    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
-            throws InvalidPartitionKeyFieldsException {
-        final List<String> partitionKeyFields = eventType.getPartitionKeyFields();
-        if (partitionKeyFields.isEmpty()) {
-            throw new RuntimeException("Applying " + this.getClass().getSimpleName() + " although event type " +
-                    "has no partition key fields configured.");
+    public String calculatePartition(final EventType eventType,
+                                     final JSONObject jsonEvent,
+                                     final List<String> partitions)
+            throws PartitioningException {
+        final var partitioningData = getPartitionKeys(eventType, jsonEvent);
+
+        return calculatePartition(partitioningData, partitions);
+    }
+
+    private String calculatePartition(final PartitioningData partitioningData, final List<String> partitions) {
+        if (partitioningData.getPartitionKeys() == null || partitioningData.getPartitionKeys().isEmpty()) {
+            throw new PartitioningException("Applying " + this.getClass().getSimpleName() + " although event type " +
```

Comment:
what about the message ? it says about event type, but here you do not check for the event type partition keys